flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/metrics/SourceReaderMetrics.java

```diff
@@ -60,4 +110,146 @@ public void recordFetchDelay(long fetchDelay) {
     public void addNumRecordsInErrors(long delta) {
         this.numRecordsInErrorsCounter.inc(delta);
     }
+
+    public void updateLastReceivedEventTime(Long eventTimestamp) {
+        if (eventTimestamp != null && eventTimestamp > 0L) {
+            lastReceivedEventTime = eventTimestamp;
+        }
+    }
+
+    public void markRecord() {
+        metricGroup.getIOMetricGroup().getNumRecordsInCounter().inc();
```

Comment:
Invoke `catchAndWarnLogAllExceptions` here.

Reply:
Ok, it has been adjusted